modules/database/src/ioc/db/dbTest.c

```diff
@@ -506,62 +506,62 @@ long dbtgf(const char *pname)
     ret_options=0;
 
     dbr_type = DBR_STRING;
-    no_elements = MIN(addr.no_elements,((sizeof(buffer))/MAX_STRING_SIZE));
+    no_elements = MIN(addr.no_elements,sizeof(buffer)/MAX_STRING_SIZE);
```

Comment:
What kind of warning does this fix? I'm still getting a warning about comparing numbers of different sign because `addr.no_elements` is signed while the result of `sizeof` isn't.

Reply:
This line did not actually have a warning. I changed it for consistency with the lines below.
Below, the gcc 11.2.0 wants to have `()` around the divisor to silence a warning.
``` ../db/dbTest.c: In function ‘dbtgf’:
../db/dbTest.c:514:57: warning: expression does not compute the number of elements in this array; element type is ‘long int’, not ‘epicsInt8’ {aka ‘signed char’} [-Wsizeof-array-div]
  514 |     no_elements = MIN(addr.no_elements,((sizeof(buffer))/sizeof(epicsInt8)));
      |                                                         ^
../db/dbTest.c:57:31: note: in definition of macro ‘MIN’
   57 | #   define MIN(x,y)  (((x) < (y)) ? (x) : (y))
      |                               ^
../db/dbTest.c:514:57: note: add parentheses around the second ‘sizeof’ to silence this warning
  514 |     no_elements = MIN(addr.no_elements,((sizeof(buffer))/sizeof(epicsInt8)));
      |                                                         ^
../db/dbTest.c:57:31: note: in definition of macro ‘MIN’
   57 | #   define MIN(x,y)  (((x) < (y)) ? (x) : (y))
      |                               ^
../db/dbTest.c:472:16: note: array ‘buffer’ declared here
  472 |     long       buffer[400];
      |                ^~~~~~
```
The compiler sees that the size of an array is divided by the size of a datatype and knows that usually people do this to caculate the number of elements. But that requires to devide by the size of the type of an array element, which is not the case here. So the compiler thinks this may be a mistake.

The existing parentheses around the dividend and the whole quotient are useless. I removed them in order to focus on the important parentheses.

Reply:
I only get signedness warnings on Windows.